components/BlurhashImage.vue

```diff
@@ -59,13 +60,63 @@ export default Vue.extend({
     };
   },
   mounted(): void {
-    if (this.item.ImageTags && this.item.ImageTags.Primary) {
-      const card = this.$refs.card as HTMLElement;
-      this.image = this.getImageUrlForElement(
-        this.item,
-        ImageType.Primary,
-        card
-      );
+    const img = this.$refs.img as HTMLElement;
+    this.image = this.getImageUrlForElement(this.item, this.type, img);
+  },
+  methods: {
+    isValidTag(): boolean {
+      if (
+        (this.item?.ImageTags &&
+          Object.prototype.hasOwnProperty.call(
+            this.item?.ImageTags,
+            this.type
+          )) ||
+        (this.type === ImageType.Backdrop &&
+          this.item?.BackdropImageTags &&
+          this.item?.BackdropImageTags.length > 0)
+      ) {
+        return true;
```

Comment:
This looks very similar to #306. In it the following is addressed:

> Changes
> - Show movie backdrop instead of primary image when in resume home section
> - Show episode primary image when in resume/up next home section
> - Fall back to backdrop if there is no thumbnail

Reply:
Worked on this before knowing anything about #306 

I haven't explained it here so much (mostly in Matrix) but I want to make this component as modular as possible with image types so, whenever we have any item object, we can pass it with an image type confidently, knowing that it will handle all the work for us. We rely too much on the mixin right now (which is still great to have in some cases), but for some basic stuff we can simplify our lives a lot by using a single component.

We can simply create an array of image types that we want to ignore, so we skip the canvas altogether and just keep the image for cases where we pass an image type where we don't want blurhashes.

In a future PR, I'll wrap most images in a LazyImage component that depends on BlurhashImage, so we can have the unloading effect we have in current web without messing too much (see #275). BlurhashImage will be the center of all of it.

Would be great to have all your input on this. cc @MrTimscampi